### PR TITLE
Create Deploy to QA.yml

### PR DESCRIPTION
**Pull request by mahtabiltarabian**
Ticket: #571

### CODE REVIEW

File: .github/workflows/Deploy to QA.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
```
Comment (dptolle) [inline .github/workflows/Deploy to QA.yml:22]: This should be node version 18.15.0

File: .github/workflows/Deploy to QA.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```
Comment (dptolle) [inline .github/workflows/Deploy to QA.yml:14]: only needs to be linux

File: .github/workflows/Deploy to QA.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+
+      - name: Install dependencies
+        run: npm install
```
Comment (dptolle) [inline .github/workflows/Deploy to QA.yml:25]: This should be npm ci instead of npm install

File: .github/workflows/Deploy to QA.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Build main.bundle.js
+        run: npm run build
```
Comment (dptolle) [inline .github/workflows/Deploy to QA.yml:28]: This should be npm run build:qa

File: .github/workflows/Deploy to QA.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    steps:
+      - name: Upload to QA using Web FTP
+        uses: SamKirkland/FTP-Deploy-Action@4.0.0
+        with:
+          server: ${{ secrets.FTP_SERVER }}
+          username: ${{ secrets.FTP_USERNAME }}
+          password: ${{ secrets.FTP_PASSWORD }}
+          local-dir: './dist'  # Modify this path to match your build output directory
+          server-dir: '/qa'     # Modify this path to the QA directory on your FTP server
```
Comment (dptolle) [inline .github/workflows/Deploy to QA.yml:42]: server-dir should be '/'

File: .github/workflows/Deploy to QA.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Upload to QA using Web FTP
+        uses: SamKirkland/FTP-Deploy-Action@4.0.0
+        with:
+          server: ${{ secrets.FTP_SERVER }}
+          username: ${{ secrets.FTP_USERNAME }}
+          password: ${{ secrets.FTP_PASSWORD }}
+          local-dir: './dist'  # Modify this path to match your build output directory
```
Comment (dptolle) [inline .github/workflows/Deploy to QA.yml:41]: local dir ist 'public' not 'dist'

**Review by dptolle — CHANGES_REQUESTED** [review]

**Comment by mahtabiltarabian** [comment]
Adding artifact to workflow(please check it)

**Pull request closed without merging**